test/framework/src/main/java/org/elasticsearch/search/aggregations/AggregatorTestCase.java

```diff
@@ -870,17 +916,24 @@ protected static DirectoryReader wrapInMockESDirectoryReader(DirectoryReader dir
     }
 
     /**
-     * Added to randomly run with more assertions on the index searcher level,
-     * like {@link org.apache.lucene.tests.util.LuceneTestCase#newSearcher(IndexReader)}, which can't be used because it also
-     * wraps in the IndexSearcher's IndexReader with other implementations that we can't handle. (e.g. ParallelCompositeReader)
+     * Creates a {@link ContextIndexSearcher} that supports concurrency running each segment in a different thread. It randomly
+     * sets the IndexSearcher to run on concurrent mode.
      */
-    protected static IndexSearcher newIndexSearcher(DirectoryReader indexReader) throws IOException {
-        if (randomBoolean()) {
-            // this executes basic query checks and asserts that weights are normalized only once etc.
-            return new AssertingIndexSearcher(random(), indexReader);
```

Comment:
you can keep using AssertingIndexSearch and just override getSlices the same as you do below for ContextIndexSearcher

Reply:
I don't think so because you need the concurrent execution in ContextIndexSearcher?

Reply:
I think it is a good idea to run AssertingIndexSearcher in non-concurrent mode

Reply:
oh right I see, you need the fix for exception handling in order to do that.